src/serac/physics/materials/hyperelastic_material.hpp

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file hyperelastic_material.hpp
+ *
+ * @brief The hyperelastic material models for the solid module
+ */
+
+#pragma once
+
+#include "mfem.hpp"
+
+#include "serac/numerics/array_4D.hpp"
+
+namespace serac {
+
+/**
+ * @brief Abstract interface class for a generic hyperelastic material
+ *
+ */
+class HyperelasticMaterial {
+public:
+  /**
+   * @brief Construct a new Hyperelastic Material object
+   *
+   */
+  HyperelasticMaterial() : Ttr_(nullptr) {}
+
+  /**
+   * @brief Destroy the Hyperelastic Material object
+   *
+   */
+  virtual ~HyperelasticMaterial() = default;
+
+  /// A reference-element to target-element transformation that can be used to
+  /// evaluate mfem::Coefficient%s.
+  /** @note It is assumed that _Ttr.SetIntPoint() is already called for the
+      point of interest. */
+
+  /**
+   * @brief Set the reference-to-target transformation. This is required to use coefficient parameters.
+   *
+   * @param[in] Ttr The reference-to-target (stress-free) transformation
+   */
+  void SetTransformation(mfem::ElementTransformation& Ttr) { Ttr_ = &Ttr; }
+
+  /**
+   * @brief Evaluate the strain energy density function, W = W(F).
+   *
+   * @param[in] F The deformation gradient
+   * @return double Strain energy density
+   */
+  virtual double EvalStrainEnergy(const mfem::DenseMatrix& F) const = 0;
+
+  /**
+   * @brief Evaluate the Cauchy stress sigma = sigma(F).
+   *
+   * @param[in] F The deformation gradient
+   * @param[out] sigma The evaluated Cauchy stress
+   */
+  virtual void EvalStress(const mfem::DenseMatrix& F, mfem::DenseMatrix& sigma) const = 0;
+
+  /**
+   * @brief Evaluate the derivative of the 1st Piola-Kirchhoff stress tensor in Voigt notation
+   * and assemble its contribution to the local gradient matrix 'A'.
+
+   * @param[in] F The deformation gradient
+   * @param[out] C Tangent moduli 4D Array in spatial form (C^e_ijkl=(d tau_ij)/(d F_km) * F_lm = J * sigma_ij delta_kl
+   + J * (d sigma_ij)/(d F_km) F_lm )
+   */
+  virtual void AssembleTangentModuli(const mfem::DenseMatrix& F, mfem_ext::Array4D<double>& C) const = 0;
```

Comment:
This name looks like it might be a carryover from MFEM. Since this function defines the tangent stiffness but doesn't integrate it like the MFEM version makes you do, what about `EvalTangentModuli` or `EvalTangentStiffness` (my preference)?